recipes/android-key-cache.md

```diff
@@ -2,49 +2,42 @@
 
 ## Description
 
-Cache project dependencies that Gradle downloads with the new key-based caching Steps, **Save Cache** and **Restore Cache**.
+Cache project dependencies that Gradle downloads with the new key-based caching Steps, **Save Gradle Cache** and **Restore Gradle Cache**.
 
 If you want to cache not only the dependencies but build tasks as well, check out [this workflow recipe](android-key-cache-build-tasks.md).
 
 ## Instructions
 
-1. Add the [Restore cache](https://github.com/bitrise-steplib/bitrise-step-restore-cache) Step to the Workflow.
-1. Add the following keys to the **Cache keys** input:
-    ```
-    gradle-cache-{{ checksum "**/*.gradle*" "**/gradle-wrapper.properties" "gradle.properties" }}
-    gradle-cache-
-    ```
-    The first key will result in a unique string based on the exact dependencies defined in your Gradle config. If there is no cache to restore with that key, the Step will move on to the second key and will restore a cache with a key that starts with `gradle-cache-`. This will result in a cache that was saved for slightly different dependencies, but it's still better than not restoring any cache.
-1. Add the usual Android Steps, such as [Android Build](https://github.com/bitrise-steplib/bitrise-step-android-build).
-1. Add the [Save cache](https://github.com/bitrise-steplib/bitrise-step-restore-cache) Step.
-    - Add `gradle-cache-{{ checksum "**/*.gradle*" "**/gradle-wrapper.properties" "gradle.properties" }}` to the **Cache key** input. The checksum at the end guarantees a new cache archive when dependencies change.
-    - Set the **Paths to cache** input to the following:
-
-    ```
-    ~/.gradle/caches
-    ~/.gradle/wrapper
-    .gradle/configuration-cache
-    ```
+1. Add the [Restore Gradle Cache](https://bitrise.io/integrations/steps/restore-gradle-cache) Step to the Workflow.
+1. Add the usual Android Steps, such as [Android Build](https://bitrise.io/integrations/steps/android-build).
+1. Add the [Save Gradle Cache](https://bitrise.io/integrations/steps/save-gradle-cache) Step.
 1. If you want to cache not only the dependencies but build tasks  as well, check out [this workflow recipe](android-key-cache-build-tasks.md).
 
+### Fine tune cache behaviour
+
+The Gradle specific cache steps use optimal cache key and path configurations maintained by Bitrise. If you want full control over what should be cached then please check out the generic [Restore Cache](https://bitrise.io/integrations/steps/restore-cache) and [Save Cache](https://bitrise.io/integrations/steps/save-cache) Steps.
+
+For example, to setup the cache key value based on your need you simply need to enter this to the **Restore Cache** and **Save Cache** step **Cache keys** input:
+```
+gradle-cache-{{ checksum "**/*.gradle*" "**/gradle-wrapper.properties" "gradle.properties" }}
```

Comment:
Both the cache key and the list of folders gained some new items since the original workflow recipe: https://github.com/bitrise-steplib/bitrise-step-save-gradle-cache/blob/main/step/step.go

Maybe it would be better to link to the code itself?

Reply:
Yeah, that is a good idea. That way the key and path examples will be always up to date.